lib/src/stream_group.dart

```diff
@@ -135,7 +154,15 @@ class StreamGroup<T> implements Sink<Stream<T>> {
   Future? remove(Stream<T> stream) {
     var subscription = _subscriptions.remove(stream);
     var future = subscription == null ? null : subscription.cancel();
-    if (_closed && _subscriptions.isEmpty) _controller.close();
+
+    if (_subscriptions.isEmpty) {
+      _onIdleController?.add(null);
+      if (_closed) {
+        onIdleController?.close();
+        _controller.close();
```

Comment:
This also sends a done event synchronously to the `_controller` stream in response to a synchronous `remove` call. I'd do `scheduleMicrotask(_controller.close);` here.